x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -124,140 +79,97 @@ export function ServiceOverviewTransactionsTable(props: Props) {
+                transactionNames,
+              },
+            },
+            isCachable: true,
+          });
 
-  const columns: Array<EuiBasicTableColumn<ServiceTransactionGroupItem>> = [
-    {
-      field: 'name',
-      name: i18n.translate(
-        'xpack.apm.serviceOverview.transactionsTableColumnName',
-        {
-          defaultMessage: 'Name',
+          return { [requestId]: metrics };
         }
-      ),
-      render: (_, { name, transactionType: type }) => {
-        return (
-          <TruncateWithTooltip
-            text={name}
-            content={
-              <TransactionDetailLink
-                serviceName={serviceName}
-                transactionName={name}
-                transactionType={type}
-                latencyAggregationType={latencyAggregationType}
-              >
-                {name}
-              </TransactionDetailLink>
-            }
-          />
-        );
-      },
-    },
-    {
-      field: 'latency',
-      name: getLatencyAggregationTypeLabel(latencyAggregationType),
-      width: px(unit * 10),
-      render: (_, { latency }) => {
-        return (
-          <SparkPlot
-            color="euiColorVis1"
-            compact
-            series={latency.timeseries ?? undefined}
-            valueLabel={asMillisecondDuration(latency.value)}
-          />
-        );
-      },
+      }
+      return fetchAggResults();
     },
-    {
-      field: 'throughput',
-      name: i18n.translate(
-        'xpack.apm.serviceOverview.transactionsTableColumnThroughput',
-        { defaultMessage: 'Throughput' }
-      ),
-      width: px(unit * 10),
-      render: (_, { throughput }) => {
-        return (
-          <SparkPlot
-            color="euiColorVis0"
-            compact
-            series={throughput.timeseries ?? undefined}
-            valueLabel={asTransactionRate(throughput.value)}
-          />
-        );
-      },
-    },
-    {
-      field: 'errorRate',
-      name: i18n.translate(
-        'xpack.apm.serviceOverview.transactionsTableColumnErrorRate',
-        {
-          defaultMessage: 'Error rate',
-        }
-      ),
-      width: px(unit * 8),
-      render: (_, { errorRate }) => {
-        return (
-          <SparkPlot
-            color="euiColorVis7"
-            compact
-            series={errorRate.timeseries ?? undefined}
-            valueLabel={asPercent(errorRate.value, 1)}
-          />
-        );
-      },
-    },
-    {
-      field: 'impact',
-      name: i18n.translate(
-        'xpack.apm.serviceOverview.transactionsTableColumnImpact',
-        {
-          defaultMessage: 'Impact',
-        }
-      ),
-      width: px(unit * 5),
-      render: (_, { impact }) => {
-        return <ImpactBar value={impact ?? 0} size="m" />;
-      },
+    // only fetches metrics when requestId changes or transaction names changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [requestId, transactionNames]
```

Comment:
Why is transactionNames a dependency here?

Reply:
If you navigate to another page, the requestId is still the same, only the transactionNames changes.

Reply:
Right, that makes sense. What about sorting them? E.g., what happens when the sort is changed on a table that only has 5 transaction groups?

Reply:
> What about sorting them? E.g., what happens when the sort is changed on a table that only has 5 transaction groups?

In that case, we'd refetch the statistics because transaction names changed its order.

Reply:
I can avoid that to happen by checking if there's only one page available. WDYT?

Reply:
I'd just sort `transactionNames`? Are there any downsides to that?

Reply:
>I'd just sort transactionNames?

Sorry, I didn't get what you mean.

Reply:
You can just call `transactionNames.sort()` before serializing. Then the order doesn't matter.

Reply:
Kibana already uses `json-stable-stringify`. Instead of using sort it's better to use that to indicate what the intention is  - we don't care that it's sorted, only that it produces the same string every time.

Reply:
> we don't care that it's sorted, only that it produces the same string every time

@sqren we do care that it's sorted when there's only one page in the table. When only one page is visible we don't have to refetch the data every time the user sorts the table.
So if I change to `json-stable-stringify` I still have to sort it to produce the same string and don't call the API again..

Reply:
nvm. I thought it was an object - I see now it's an array of objects.